src/main/kotlin/org/rust/ide/annotator/RsExpressionAnnotator.kt

```diff
@@ -92,10 +94,27 @@ private class RedundantParenthesisVisitor(private val holder: RsAnnotationHolder
     }
 
     private fun RsExpr?.warnIfParens(message: String) {
-        if (this !is RsParenExpr) return
-        val fix = RemoveParenthesesFromExprIntention()
-        if (fix.isAvailable(this))
-            holder.createWeakWarningAnnotation(this, message)?.registerFix(RemoveParenthesesFromExprIntention())
+        if (this !is RsParenExpr || !canWarn(this)) return
+        val fix = RemoveRedundantParenthesesFix(this)
+        val descriptor = InspectionManager.getInstance(project)
+            .createProblemDescriptor(this, message, fix, ProblemHighlightType.GENERIC_ERROR_OR_WARNING, true)
+        holder.createWeakWarningAnnotation(this, message)?.registerFix(fix, null, null, descriptor)
```

Comment:
Oh, sorry - I forget that `Annotation` has more complex API for `LocalQuickFix` than for `IntentionAction`.
Let's revert replacing `LocalQuickFixAndIntentionActionOnPsiElement` with `LocalQuickFixOnPsiElement` and simplify this code in the corresponding way